lib/src/components/NxDrawer/NxDrawer.tsx

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { useEffect, useRef } from 'react';
+import classnames from 'classnames';
+
+import AbstractDialog from '../AbstractDialog/AbstractDialog';
+import NxCloseButton from '../NxCloseButton/NxCloseButton';
+import withClass from '../../util/withClass';
+
+import { Props, propTypes } from './types';
+
+import './NxDrawer.scss';
+
+const DRAWER_OPEN_CLASS_NAME = 'nx-drawer--open';
+
+const _NxDrawer = (props: Props) => {
+  const {
+    className,
+    onCancel,
+    children,
+    variant,
+    headerTitle,
+    headerSubtitle,
+    headerParagraph,
+    ...attrs
+  } = props;
+
+  const dialogRef = useRef<HTMLDialogElement>(null);
+
+  useEffect(() => dialogRef.current?.classList.add(DRAWER_OPEN_CLASS_NAME), [dialogRef]);
+
+  const closeDrawer = () => {
+    const prefersReducedMotion = window.matchMedia('(prefers-reduced-motion: reduce)').matches;
+    dialogRef.current?.classList.remove(DRAWER_OPEN_CLASS_NAME);
+    if (prefersReducedMotion) {
+      onCancel();
+    }
+  };
+
+  const handleTransitionEnd = () => {
+    if (!dialogRef.current?.classList.contains(DRAWER_OPEN_CLASS_NAME)) {
```

Comment:
Going along with https://github.com/sonatype/sonatype-react-shared-components/pull/736/files?short_path=c557c3f#r939173115, there should be a piece of state we check here rather than inspecting the DOM